tests/integration/targets/aci_hsrp_interface_policy/tasks/main.yml

```diff
@@ -0,0 +1,154 @@
+# Author: Shreyas Srish (@shrsr)
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# SET VARS
+- name: Set vars
+  set_fact:
+    aci_info: &aci_info
+      host: '{{ aci_hostname }}'
+      username: '{{ aci_username }}'
+      password: '{{ aci_password }}'
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: '{{ aci_output_level | default("info") }}'
+
+# CLEAN ENVIRONMENT
+- name: Remove test tenant before we kickoff
+  cisco.aci.aci_tenant: &tenant_absent
+    <<: *aci_info
+    tenant: ansible_test
+    state: absent
+
+- name: Verify Cloud and Non-Cloud Sites in use.
+  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- name: Execute tasks only for non-cloud sites
+  when: query_cloud.current == []  # This condition will execute only non-cloud sites
+  block:  # block specifies execution of tasks within, based on conditions
+  # SETUP ENVIRONMENT
+  - name: Create tenant
+    cisco.aci.aci_tenant: &tenant_present
+      <<: *tenant_absent
+      state: present
+
+  - name: Add a HSRP interface policy in check mode
+    cisco.aci.aci_hsrp_interface_policy: &hsrp_present
+      <<: *tenant_present
+      hsrp: hsrp1
+      controls: ["bia"]
+      reload_delay: 2
+      delay: 2
+      state: present
+    check_mode: true
+    register: cm_add_hsrp
+
+  - name: Add a HSRP interface policy in normal mode
+    cisco.aci.aci_hsrp_interface_policy: *hsrp_present
+    register: nm_add_hsrp
+
+  - name: Verify add hsrp
+    assert:
+      that:
+      - cm_add_hsrp is changed
+      - nm_add_hsrp is changed
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.ctrl == "bia"
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.delay == "2"
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.reloadDelay == "2"
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.name == "hsrp1"
+      - nm_add_hsrp.current.0.hsrpIfPol.attributes.dn == "uni/tn-ansible_test/hsrpIfPol-hsrp1"
```

Comment:
Should we check proposed and previous also like we do in other modules? Comment applies for other modules also.

Reply:
Ok